### DISCUSSION TITLE
Loading debug symbols from musllibc

### DISCUSSION THREAD

**Discussion by astevins** [sid-agrawal/OSmosis#21] [Knowledge Base]
This is an info post if you want to use debug symbols for musllibc in GDB.

1. Enable musllibc debug: in `/projects/musllibc/makefile` line 57, change `${ENABLE_DEBUG}` to `--enable-debug`. I have not figured out how to properly set the flag, as build errors occured when I tried to set it from the cmake file.
2. Extract the desired object file: eg. to extract malloc, use `ar -xv apps/sel4test-driver/musllibc/build-temp/lib/libc.a malloc.o` OR `ar -xv apps/sel4test-driver/musllibc/build-temp/stage/lib/libc.a malloc.o` (whichever works) from the build folder.
3. Now the symbols should be loaded automatically in GDB.

**Comment by sid-agrawal** [comment]
Thanks for documenting it!
